store/tikv/2pc.go

```diff
@@ -411,6 +413,16 @@ func (c *twoPhaseCommitter) primary() []byte {
 	return c.primaryKey
 }
 
+func (c *twoPhaseCommitter) secondaries() [][]byte {
+	if !config.GetGlobalConfig().TiKVClient.EnableParallelCommit || len(c.mutations.keys) == 0 {
+		return [][]byte{}
+	}
+	if bytes.Compare(c.mutations.keys[0], c.primary()) == 0 {
```

Comment:
The transaction primary key may not be the first key in keys array, https://github.com/pingcap/tidb/blob/master/store/tikv/2pc.go#L361,
maybe we need to switch it in the `initKeysAndMutations` function

Reply:
Does this code: https://github.com/pingcap/tidb/blob/master/store/tikv/2pc.go#L545-L549 not mean it will always be first?

Reply:
L545-L549 means the first `batch` contains primary key and it will be processed firstly sending RPCs, the `batchMutations` are results grouped by key regions and split by batch size limit. But after `initKeysAndMutations`, it's not guaranteed the primary key will be the first key in `c.mutations.Keys`.

Reply:
Thanks!